src/KingSystem/Map/mapObject.cpp

```diff
@@ -1,7 +1,373 @@
+
+    auto* data = &PlacementMgr::instance()->mPlacementActors->mActorData[mActorDataIdx];
+    f32 dist = 0.0;
+
+    if (data->mFlags.isOnBit(ActorData::Flag::TraverseDistReset))
+        return 0.0;
+
+    if (data->mFlags.isOnBit(ActorData::Flag::TraverseDistReset))
+        dist = 0.0;
+
+    else if (data->mFlags.isOnBit(DFlag::TraverseDist400) ||
+             data->mFlags.isOnBit(DFlag::TraverseDist200) ||
+             data->mFlags.isOnBit(DFlag::TraverseDist4000) ||
+             data->mFlags.isOnBit(DFlag::TraverseDist800))
+        dist = mTraverseDist + 0.0;
+
+    else if (data->mFlags.isOffBit(DFlag::TraverseDist2000))
+        dist = mTraverseDist + 0.0;
+    else
+        dist = mTraverseDist;
+
+    f32 mult = getDispDistanceMultiplier();
+
+    if (dist > 0.0)
+        return dist * mult;
+    else
+        return 100.0 * mult;
+}
+
+sead::Vector3f Object::getScale() const {
+    sead::Vector3f vec;
+
+    mMubinIter.getScale(&vec);
+    return vec;
+}
+
+const char* Object::getUnitConfigName() const {
+    auto* mgr = PlacementMgr::instance();
+
+    if (mgr != nullptr && mgr->mPlacementActors != nullptr)
+        return mgr->mPlacementActors->mActorData[mActorDataIdx].mActorName.cstr();
+
+    return getUnitConfigNameFromByaml();
+}
+
+const char* Object::getUnitConfigNameFromByaml() const {
+    const char* result;
+    if (!mMubinIter.tryGetParamStringByKey(&result, "UnitConfigName"))
+        result = "";
+
+    return result;
+}
+
+Object* Object::findPlacementLODLinkObject(const PlacementActors*) const {
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+Object* Object::findPlacementLODLinkObject() const {
+    if (!mLinkData)
+        return nullptr;
+
+    ObjectLink* result = mLinkData->mLinksToSelf.findLinkWithType(MapLinkDefType::PlacementLOD);
+    if (result)
+        return result->other_obj;
+
+    return nullptr;
+}
+
+const char* Object::getHashIdStringDebug() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+const char* Object::getHashIdStringDebug_0() const {
+    act::ActorDebug::HashUnused t;
+
+    const char* terminator;
+
+    if (mMubinIter.tryGetParamUIntByKey(&t.hash, "HashId")) {
+        terminator = act::ActorDebug::instance()->getNullStr(&t);
+    } else
+        terminator = "";
+
+    return terminator;
+}
+
+void* Object::getRails() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+void* Object::getRails_0() const {
+    if (!mLinkData)
+        return nullptr;
+
+    return mLinkData->mRails;
+}
+
+bool Object::allocLinkData(sead::ExpHeap* heap) {
```

Comment:
```suggestion
bool Object::allocLinkData(sead::Heap* heap) {
```

in the IDB you'll often see sead::ExpHeap* when it should be sead::Heap*